File: apps/api/src/data_intelligence_hub/schemas/project.py
from __future__ import annotations

import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

ProjectDomain = Literal["osint", "ecommerce", "social", "competitor", "mixed"]
ProjectStatus = Literal["active", "archived"]


class ProjectCreateRequest(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    description: str | None = Field(default=None, max_length=5000)
    domain: ProjectDomain


class ProjectUpdateRequest(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=200)
    description: str | None = Field(default=None, max_length=5000)
    domain: ProjectDomain | None = None
    status: ProjectStatus | None = None


class ProjectResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    workspace_id: uuid.UUID
    name: str
    description: str | None
    domain: str
    status: str
    owner_id: uuid.UUID
    created_at: datetime
    updated_at: datetime

File: apps/web/src/types/project.ts
export type AuthSession = {
  user: {
    id: string;
    email: string;
    name: string;
    status: string;
  };
  workspace: {
    id: string;
    name: string;
    slug: string;
    ownerId: string;
  };
};

export type ProjectDomain = "osint" | "ecommerce" | "social" | "competitor" | "mixed";
export type ProjectStatus = "active" | "archived";

export type Project = {
  id: string;
  name: string;
  description: string | null;
  domain: ProjectDomain;
  status: ProjectStatus;
  intelligenceCount: number;
  sourceCount: number;
};

export type ProjectCreateInput = {
  name: string;
  description?: string;
  domain: ProjectDomain;
};
